src/jcvi/compara/synteny.py

```diff
@@ -62,13 +77,49 @@ def __init__(self, filename, defaultcolor="#fb8072", header=False, allow_hex=Fal
             else:
                 hl = False
 
+            # Remove trailing newline and split by tab
             atoms = row.rstrip().split("\t")
+            # Strip whitespace from each atom
             atoms = [x.strip() for x in atoms]
+            # Replace empty strings with "."
             atoms = ["." if x == "" else x for x in atoms]
+
+            # Check atoms for spaces
+            if any(" " in x for x in atoms):
+                logger.error(
+                    "Spaces found in row `{}` of blocks file. Columns must be tab delimited.\n{}".format(
+                        row_count, row.strip()
+                    )
+                )
+                exit(1)
+
+            # Warning for empty lines
+            if not atoms or all(x == "." for x in atoms):
+                logger.warning(
+                    "Empty line or all columns are empty in row `{}` of blocks file.".format(
+                        row_count
+                    )
+                )
+                continue
+
+            # Truncate or pad the atoms to match ncols
             if len(atoms) > ncols:
+                logger.warning(
+                    "Row length exceeds expected columns in blocks file, truncating to {} at line {}:\n {}".format(
+                        ncols, row_count, row.strip()
+                    )
+                )
+                # Truncate to ncols
                 atoms = atoms[:ncols]
             elif len(atoms) < ncols:
+                logger.warning(
+                    "Row length less than expected columns in file `{}`, padding to {} cols at line {}:\n{}".format(
+                        filename, ncols, row_count, row.strip()
+                    )
+                )
+                # Pad with "." to match ncols
                 atoms = atoms + ["."] * (ncols - len(atoms))
```

Comment:
@tanghaibao I've added some warnings around the existing behaviour (pad or truncate).

I think it would be cleaner to raise an error if any line has a column count that differs from the first row (taken as correct number).